File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
public class Solution {
    public int maxProfit(int prices[]) 
    {
        int minPrice = prices[0];
        int maxProfit = 0;
        for (int i = 0; i < prices.length; i++) 
        {
             // Update max_profit if the difference between current price and min_price is greater than current max_profit
            maxProfit = Math.max(maxProfit, prices[i]-minPrice);
            // Update min_price if the current price is less than the current min_price
            minPrice = Math.min(prices[i], minPrice);
            // if (prices[i] < minprice)
            //     minprice = prices[i];
            // else if (prices[i] - minprice > maxprofit)
            //     maxprofit = prices[i] - minprice;
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    n := len(prices)
    if n == 0 {
        return 0
    }

    minPrice := prices[0]
    maxProfit := 0

    for i := 1; i < n; i++ {
        if prices[i] - minPrice > maxProfit {
            maxProfit = prices[i] - minPrice
        }
        if prices[i] < minPrice {
            minPrice = prices[i]
        }
    }

    return maxProfit
}